crates/runtime/src/model/tool_use.rs

```diff
@@ -312,6 +312,9 @@ impl ToolUsingChat {
             req
         } else {
             runtime_tools.extend(req.tools.clone().unwrap_or_default());
+            // Ensure function names are unique. Tool-use recursion sometimes creates duplicates.
+            runtime_tools.sort_by(|a, b| a.function.name.cmp(&b.function.name));
```

Comment:
In some recursions, tools would be duplicated. Anthropic doesn't like this (Openai doesn't care).